app/renderer/components/task_suite.js

```diff
@@ -71,7 +90,9 @@ module.exports = {
         },
         showAddTab() {
             return AppStatus.editMode && this.suite.length < AppStatus.maxTasksPerSuite;
+        },
+        orderTasks() {
+            return this.suite.tasks.sort((a, b) => a.order - b.order);
```

Comment:
Instead of using the order property of a task, why not just using an array?

Reply:
not quite sure what do you mean here ...

Reply:
So, atm you have a set of tasks, and within each task you have a explicit property (order), and you use it to well... order.

However, it doesn't make sense to have that property as part of a task (why would a task care of its own position within different tasks?). In fact, a suite has an array/list of tasks, which, by definition, maintains the order (that was done by design at the beginning). So:

Instead of having:
```js
[{
   name: "second task",
   order: 2
}, {
   name: "first task",
   order: 1
}]
```

And ordering them according to its internal value. We could simply take care in maintaining the array order when writing, reading, moving tasks and rendering:
```js
[{
  name: "first task"
}, {
  name: "second task"
}]
```

We do not need to manually store the order and sort according to it. The array data structure already do that for us, doing this way has some advantages:
* No order property in tasks, no manually ordering each time, the array already contains this information, in general, simpler
* Better for storing/reading data, as we do not need to store the ordering property
* It doesn't make sense to store order in the tasks as it is irrelevant for it, moving a task from one place to another shouldn't change the task itself, but its container (suite)

Basically, I have the impression that solving this issue doesn't require changes to the task.js class, as everything should be part of the suite

I'm not sure if I'm explaining myself, sorry, if you have any doubt I'm happy to make a simple prototype/example to show my point

From what I saw in your code it shouldn't be hard to change it to the way I see it, as most changes you made in the suite are well defined, clean and simple enough to adapt them